src/oop/user_defined_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,116 @@
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void dishWasherClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + testClassName + " is missing a required field";
+        String incorrectModifierMessage = "One of your fields in the " + testClassName +
+                " class does not have the correct visibility modifier.";
+        assertTrue(classInstance.hasField(intAttributeName1, int.class),
+                missingFieldMessage);
+        assertTrue(classInstance.hasModifier(intAttributeName1, "private"),
+                incorrectModifierMessage);
+        assertTrue(classInstance.hasField(intAttributeName2, int.class),
+                missingFieldMessage);
+        assertTrue(classInstance.hasModifier(intAttributeName2, "private"),
+                incorrectModifierMessage);
+    }
+
+    @Test
+    public void dishWasherClassHasRequiredConstructor() {
+        String missingConstructorMessage = "Your " + testClassName + " class is missing a required constructor.";
+        Class<?>[] arguments= {
+                int.class, int.class
+        };
+        assertTrue(classInstance.hasConstructor(arguments), missingConstructorMessage);
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(35, 12),
+                Arguments.of(478273, 723662367)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void dishWasherConstructorInitializesFieldsCorrectly(int attribute1, int attribute2) throws Throwable{
+        Object[][] arguments = {
+                {attribute1, int.class},
+                {attribute2, int.class}
+        };
+        Object instance = classInstance.createInstance(arguments);
+        String incorrectFieldInstantiationMessage =
+                "Your " + testClassName + " constructor does not correctly initialize the object." ;
+        _assertEquals(attribute1, classInstance.getFieldValue(instance, intAttributeName1),
+                incorrectFieldInstantiationMessage);
+        _assertEquals(attribute2, classInstance.getFieldValue(instance, intAttributeName2),
+                incorrectFieldInstantiationMessage);
+    }
+
+    private static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(2, 3, 1, 1, 1, 2),
+                Arguments.of(15, 33, 14, 15, 1, 18),
+                Arguments.of(378, 32891, 378, 32891, 0, 0),
+                Arguments.of(0, 0, 0, 0, 0, 0),
+                Arguments.of(5, 10, 0, 1, 5, 9)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    public void correctLoadDishWasherMethod(int attribute1, int attribute2, int remove1, int remove2, int remainder1, int remainder2) throws Throwable{
+        Object[][] arguments = {
```

Comment:
I don't think you actually need the remainders here as parameters you can check for attribute - remove

Reply:
Yes, but it effectively achieves the same end and I prefer to minimize computer calculation when possible. Removing the remainder attributes and swapping them out for a calculation doesn't improve anything I don't believe.